lua/telescope/_extensions/loogle/loogle_search.lua

```diff
@@ -0,0 +1,101 @@
+local pickers = require('telescope.pickers')
+local actions_state = require 'telescope.actions.state'
+local finders = require('telescope.finders')
+local conf = require('telescope.config').values
+local actions = require('telescope.actions')
+local previewers = require('telescope.previewers')
+
+local curl = require 'plenary.curl'
+
+local M = {}
+
+M.ask_for_type = function()
+  local type = ''
+  local on_confirm = function(input) type = input end
+  vim.ui.input({prompt = 'Enter Loogle type pattern: '}, on_confirm)
+  if type == nil or type == '' then
+    return nil, 'Type was empty'
+  end
+  return type
+end
+
+M.look_for_type = function(type)
+  local res = curl.get {
+    url = 'https://loogle.lean-lang.org/json',
+    query = { q = type },
+    headers = {
+      User_Agent = 'lean.nvim'
+    },
+    accept = 'application/json',
+  }
+
+  if res.status ~= 200 then
+    return nil, 'Loogle returned status code: ' .. res.status
+  end
+
+  local body = vim.fn.json_decode(res.body)
+  if body.error then
+    return nil, 'Loogle returned error: ' .. body.error
+  end
+
+  if body.count == 0 then
+    return nil, 'Loogle found no matches'
+  end
+
+  return body.hits, nil
+end
+
+M.find = function(telescope_opts)
+  telescope_opts = vim.tbl_extend('keep', telescope_opts or {}, {})
+
+  local type, results, err
+
+  type, err = M.ask_for_type()
+  if err then
+    print(err)
```

Comment:
Probably instead of printing for these we should use `vim.ui.notify` with an `error` log level.

Reply:
`vim.ui.notify` doesn't seem to exist on my Neovim, is this a nightly feature?

Reply:
It's `vim.notify`, sorry.